internal/bridge/crunchybridgecluster/watches.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 - 2023 Crunchy Data Solutions, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crunchybridgecluster
+
+import (
+	"context"
+
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
+)
+
+// watchForRelatedSecret handles create/update/delete events for secrets,
+// passing the Secret ObjectKey to findCrunchyBridgeClustersForSecret
+func (r *CrunchyBridgeClusterReconciler) watchForRelatedSecret() handler.EventHandler {
+	handle := func(secret client.Object, q workqueue.RateLimitingInterface) {
+		ctx := context.Background()
```

Comment:
That's a good question. This was done by @benjaminjb  in the original iteration of the Bridge integration, so I'm not sure if there was specific reason for not using the global context...

Reply:
(The contents of the `watches.go` file is "green" simply because I moved these functions from `crunchybridgecluster_controller.go` to their own file.)

Reply:
Yeah, I understand that this code is copied over. I just noticed it while I was looking through. I see other places where we use a Background context, so this is probably fine.

Reply:
Okay, I looked at this a bit and `watchForRelatedSecret` is called by `SetupWithManager` which does not define a context and doesn't accept one as an argument. None of our other `SetupWithManager` functions in our other controller packages have a context defined either. That doesn't necessarily mean we shouldn't do it here, but I presume that is why it was done this way.

Reply:
Also, similarly in `postgrescluster`, `controllerRefHandlerFuncs` is called in `SetupWithManager` and it defines its own context rather than having one passed in as an argument.

Reply:
If we ever do want a context we can add it in :+1: